File: questions/Python/Basic/Es3.py
# -- "Normal", weight: '50'
# 'Realizza una funzione "run" '
# 'che sottragga il primo con il secondo.'
# '********************** '
# '********************** '
# '********************** '
# '********************** '
# '********************** '
# '********************** '
# '********************** '

import unittest
from RunningFile import run


class Test(unittest.TestCase):
    def test_1(self):
        self.assertEqual(run(1, 2), -1, 'Errore')

    def test_2(self):
        self.assertEqual(run(0, 2), -2, 'Errore')

    def test_3(self):
        self.assertEqual(run(-1, 2), -3, 'Errore')

    def test_4(self):
        self.assertEqual(run(5, 2), 3, 'Errore')

    def test_5(self):
        self.assertEqual(run(100, 99), 1, 'Errore')


unittest.main()

File: questions/Java/Collection/Es3.java
/* -- "Normal", weight: '1'
/* 'Realizza una classe Esercizio a cui si passi nel costruttore due numeri interi, ' */
/* 'tale classe deve avere una funzione run che ritorni il primo diviso il secondo.' */


public class Es3{
    public static void main(String[] args) {
        Esercizio e1 = new Esercizio(1,2);
        Esercizio e2 = new Esercizio(2,-2);
        if (e1.run() == 0.5){
            if (e2.run() == -1){
                System.out.println("..\nTest eseguiti con successo.");
            }
            else{
                System.out.println(".F\nUn test fallito.");
            }
        }
        else {
            System.out.println("FF\nTest falliti.");
        }
    }
}